src/main/java/io/neonbee/data/DataQuery.java

```diff
@@ -186,7 +188,9 @@ public DataQuery(DataAction action, String uriPath, String query, Map<String, Li
     public DataQuery(DataAction action, String uriPath, String query, Map<String, List<String>> headers, Buffer body) {
         this.action = action;
         this.setUriPath(uriPath); // Calling the setter here, because there is an additional check implemented.
-        this.headers = CollectionHelper.mutableCopyOf(headers != null ? headers : Map.of());
+        if (headers != null) {
+            this.headers.putAll(headers);
+        }
```

Comment:
You create only a shallow copy of the header map. This breaks the copy method. This way, you could create a copy using the copy method and change the values of the two DataQuery objects by changing them in one object.

Example test case:
```
DataQuery query1 = new DataQuery("uri", "name=Hodor", Map.of("header1", new ArrayList<>(List.of("value1"))));
assertThat(query1.getHeaderValues("header1")).hasSize(1);
DataQuery query2 = query1.copy();
query2.getHeaderValues("header1").add("value2");
assertThat(query1.getHeaderValues("header1")).hasSize(1);
```